### PR TITLE
feat: component, icon, export, docs

### PR DESCRIPTION
**Pull request by jessesoomy**

### CODE REVIEW

File: packages/react/src/components/dialog.tsx
Comment (irfancoder) [inline packages/react/src/components/dialog.tsx:1]: Dialog isnt up to spec. TODO
1. `DialogTitle` & `DialogDescription` incorrect font size & weight
2. `DialogClose` missing border & rounded (think can use same spec `Button.variant="default-outline"`
3. `DialogFooter` optional `border-t` 

Refer to Figma: https://www.figma.com/design/svmWSPZarzWrJ116CQ8zpV/MYDS-(Beta)?node-id=476-7024&node-type=instance&t=oijp5bsFSusOoHwK-0

File: apps/storybook/stories/dialog.stories.tsx
Comment (irfancoder) [inline apps/storybook/stories/dialog.stories.tsx:1]: Include dark mode story

**Review by irfancoder — CHANGES_REQUESTED** [review]

File: packages/react/src/components/dialog.tsx
Comment (jessesoomy) [reply packages/react/src/components/dialog.tsx:1]: @irfancoder Patched.

File: apps/storybook/stories/dialog.stories.tsx
Comment (jessesoomy) [reply apps/storybook/stories/dialog.stories.tsx:1]: Patched as well.

**Pull request merged**